sycamore/transforms/split_elements.py

```diff
@@ -0,0 +1,112 @@
+from ray.data import Dataset
+
+from sycamore.data import Document, Element
+from sycamore.functions.tokenizer import Tokenizer
+from sycamore.plan_nodes import Node, Transform, SingleThreadUser, NonGPUUser
+from sycamore.transforms.map import generate_map_function
+
+
+class SplitElements(SingleThreadUser, NonGPUUser, Transform):
```

Comment:
Can we do the OOP thing and turn this into an ElementSplitter or some such interface and then subclass as something like the `RecursiveDivisionElementSplitter`